girder/api/access.py

```diff
@@ -80,3 +77,15 @@ def accessDecorator(*args, **kwargs):
         return fun(*args, **kwargs)
     accessDecorator.accessLevel = 'token'
     return accessDecorator
+
+
+def cookie(fun):
+    """
+    REST endpoints that allow the use of a cookie for authentication should be
+    wrapped in this decorator.
+
+    This should only be set on endpoints that are read-only (GET requests),
+    to avoid vulnerabilities to Cross-Site Request Forgery attacks.
```

Comment:
On a long-lingering local branch of mine, I implemented such decorators for the access levels. They look like this:

```
def admin(*args, **kwargs):
    """
    Functions that require administrator access should be wrapped in this
    decorator.

    :param scope: To also expose this endpoint for certain token scopes,
        pass those scopes here. If multiple are passed, all will be required.
    :type scope: str or list of str
    """
    if len(args) == 1 and callable(args[0]):  # Raw decorator
        @six.wraps(args[0])
        def wrapped(*iargs, **ikwargs):
            rest.requireAdmin(rest.getCurrentUser())
            return args[0](*iargs, **ikwargs)
        wrapped.accessLevel = 'admin'
        return wrapped
    else:  # We should return a decorator
        def dec(fun):
            @six.wraps(fun)
            def wrapped(*iargs, **ikwargs):
                token = rest.getCurrentToken()
                if not _tokenModel.hasScope(token, kwargs.get('scope')):
                    rest.requireAdmin(rest.getCurrentUser())
                return fun(*iargs, **ikwargs)
            wrapped.accessLevel = 'admin'
            return wrapped
        return dec
```

Hopefully that helps :)